From fcb38631aaf9b937e7916f327e07959aa1eaf2a3 Mon Sep 17 00:00:00 2001
From: Shoaib Ahmed <shoahme@qti.qualcomm.com>
Date: Mon, 30 Mar 2026 15:44:30 +0530
Subject: [PATCH 37/40] qcacld-3.0: Add check to avoid buffer overflow

Currently in api cm_roam_stats_print_11kv_info(), out of bound write can
possibly happen.

To avoid this add check to avoid buffer overflow.

CRs-Fixed: 4483061
Change-Id: Ibfefc4f4af06ef5eea1b700dcf3838df526a2776
---
 .../connection_mgr/dispatcher/src/wlan_cm_roam_api.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c b/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c
index 4470db4890..5cf6e06742 100644
--- a/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c
+++ b/components/umac/mlme/connection_mgr/dispatcher/src/wlan_cm_roam_api.c
@@ -3825,13 +3825,23 @@ cm_roam_stats_print_11kv_info(struct wlan_objmgr_psoc *psoc,
 
 	tmp = buf;
 	if (num_ch) {
-		buf_cons = qdf_snprint(tmp, buf_left, "{ ");
+                buf_cons = qdf_snprint(tmp, buf_left, "{ ");
 		buf_left -= buf_cons;
 		tmp += buf_cons;
 
 		for (i = 0; i < num_ch; i++) {
+			if (!buf_left) {
+				mlme_err("buf_left is empty");
+				qdf_mem_free(buf);
+				return;
+			}
 			buf_cons = qdf_snprint(tmp, buf_left, "%d ",
 					       neigh_rpt->freq[i]);
+			if (!buf_cons || buf_cons > buf_left) {
+				mlme_err("buf_cons is empty");
+				qdf_mem_free(buf);
+				return;
+			}
 			buf_left -= buf_cons;
 			tmp += buf_cons;
 		}
-- 
2.34.1

